src/actions/helpers/index.ts

```diff
@@ -0,0 +1,21 @@
+import { BigNumber, BigNumberish } from "ethers";
+import { Registrar } from "../../contracts/types";
+
+export const validateOwner = async (
+  domainId: BigNumberish,
+  potentialOwner: string,
+  registrar: Registrar,
+  message: string
+) => {
+  const owner = await registrar.ownerOf(domainId);
+  if (potentialOwner !== owner) throw Error(message);
+};
+
+export const validateStatus = async (
```

Comment:
Nit: misleading function name. It's unclear what "status" is without looking at this function. Perhaps `validateDomainUnlocked` or `ensureDomainUnlocked` is a more clear name.